spark/dl/src/main/scala/com/intel/analytics/bigdl/example/udfpredictor/README.md

```diff
@@ -0,0 +1,496 @@
+    ```
+
+4. Run this command to generate test data for prediction in Structured Streaming:
+    ```bash
+    usage: create_test_texts.py [-h] [-s SRC] [-t TEST]
+    
+    Create test text files from source
+    
+    optional arguments:
+     -h, --help            show this help message and exit
+     -s SRC, --src SRC     source directory
+     -t TEST, --test TEST  test directory
+     
+    Example: python ./create_test_texts.py -s BASE_DIR/20_newsgroup -t BASE_DIR/test
+       
+    ```  
+    
+5. Run Spark SQL example
+
+* Training the model and make predictions.
+   
+   Run the commands:
+   
+   If you do not have the pre-trained model, you need to use this command to train the model and use this model to predict text classification of text records with UDF.
+
+   Example:
+
+   ```shell
+   BASE_DIR=${PWD} # where is the data
+   MASTER=loca[*] # the master url
+   ./dist/bin/bigdl.sh -- spark-submit --master $MASTER --driver-memory 20g \
+       --class com.intel.analytics.bigdl.example.modeludf.BatchPredictor \
+         ./dist/lib/bigdl-0.1.0-SNAPSHOT-jar-with-dependencies.jar \
+       -c 4 \
+       -n 1 \
+       --batchSize 32 \
+       --baseDir $BASE_DIR \
+       --partitionNum 4 \
+       --checkpoint  $BASE_DIR/model/text \
+       --dataDir $BASE_DIR/test
+   ```
+        
+   In the above commands, 
+   
+   -c: How many cores of your machine will be used in the training. Note that the core number should be physical core number. If your machine turn on hyper threading, one physical core will map to two OS core.
+   
+   -n: Node number.
+   
+   --batchSize: how many text files to be trained at one time
+   
+   -baseDir: folder containing trainning text files and word2Vec embedding.
+   
+   --partitionNum: number to partition training data
+   
+   --checkpoint: location to save model
+   
+   --dataDir: Directory containing the test data
+
+   If you have saved model, you need to use this command to predict text classification of text records with UDF.
+      
+   Example:
+     
+   ```shell
+   BASE_DIR=${PWD} # where is the data
+   MASTER=loca[*] # the master url
+   ./bigdl.sh -- spark-submit --master $MASTER --driver-memory 5g \
+              --class com.intel.analytics.bigdl.example.modeludf.BatchPredictor \
+                bigdl-0.1.0-SNAPSHOT-jar-with-dependencies.jar \
+              -c 4 \
+              -n 1 \  
+              --baseDir $BASE_DIR \
+              --modelPath  $BASE_DIR/model/text/model.1 \
+              --dataDir $BASE_DIR/test
+   ```
+   In the above commands, 
+   
+   -c: How many cores of your machine will be used in the training. Note that the core number should be physical core number. If your machine turn on hyper threading, one physical core will map to two OS core.
+   
+   -n: Node number.
+   
+   -baseDir: folder containing trainning text files and word2Vec embedding 
+   
+   --modelPath: model location
+   
+   --dataDir: Directory containing the test data
+        
```

Comment:
To run spark cluster, user only need to change the maser url. Do we really need another scripts?